float32[] joints
float32 x
float32 y
float32 z